.github/workflows/ci.yml

```
@@ -712,6 +305,13 @@ jobs:
           repository: cppalliance/capy
           ref: develop
 
+      - name: Clone Boost.Corosio
+        uses: actions/checkout@v3
```

actionlint (1.7.10)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)